tests/no-learning/test_DDM.py

```diff
@@ -61,3 +64,30 @@ def test_DDM():
         # if you do not specify, assert_allcose will use a relative tolerance of 1e-07,
         # which WILL FAIL unless you gather higher precision values to use as reference
         np.testing.assert_allclose(val, expected, atol=1e-08, err_msg='Failed on expected_output[{0}]'.format(i))
+
+class TestDDMCorrectness:
+    @pytest.mark.parametrize("mode", ['Python',
+                                      pytest.param('LLVM', marks=pytest.mark.llvm),
+                                      pytest.param('LLVMExec', marks=pytest.mark.llvm),
+                                      pytest.param('LLVMRun', marks=pytest.mark.llvm),
+                                      pytest.param('PTXExec', marks=[pytest.mark.llvm, pytest.mark.cuda]),
+                                      pytest.param('PTXRun', marks=[pytest.mark.llvm, pytest.mark.cuda])
+                                      ])
+    @pytest.mark.parametrize(["noise", "threshold", "expected_results"],[
+                             (1.0, 0.0, (0.0, 1.0)),
+                             (1.5, 2, (-2.0, 1.0)),
+                             (10.0, 10.0, (10.0, 29.0)),
+                             (100.0, 100.0, (100.0, 76.0)),
+                            ])
+    def test_ddm_is_finished(self, mode, noise, threshold, expected_results):
+        comp = Composition()
+        ddm = DDM(execute_until_finished=True,
+                    function=DriftDiffusionIntegrator(threshold=threshold, noise=noise))
+        comp.add_node(ddm)
+
+        results = comp.run([0], bin_execute=mode)
+
+        # FIXME: Results differ in shape between compiled and Python mode
+        results = np.array(results).flatten()
+        for result, expected_result in zip(results, expected_results):
+            assert result == expected_result
```

Comment:
use the same workaround as other DDM tests. there's no reason to have multiple versions of the same thing.